### PR TITLE
fix issue with contig_bin.tsv reading #67

### PR DESCRIPTION
**Pull request by brambloemen**
fix for issue #67

### CODE REVIEW

File: nanomotif/binnary/data_processing.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def load_data(args):
     bin_motifs = pl.read_csv(args.bin_motifs, separator="\t")
     
     # Polars automatically infers headers; if args.contig_bins file doesn't have headers, you need to specify it
-    contig_bins = pl.read_csv(args.contig_bins, separator="\t", has_header=False) \
```
Comment (brambloemen) [inline nanomotif/binnary/data_processing.py:26]: default infer_schema_length is 100, might be too low if many bins are named as either string or integer (e.g. in the case of ensemble of several different binning methods)

File: nanomotif/binnary/utils.py [hunk 2]
```diff
@@ -2,15 +2,20 @@
...
 def split_bin_contig(df):
-    df = df \
-        .with_columns(
-            pl.when(pl.col("bin_compare").is_not_null()).then(
-                pl.col("bin_compare").str.replace(pl.col("contig_bin").fill_null("").first() + "_", "", literal = True)
-            ).over("bin_compare").alias("contig")
-        )
-        
-        
-
```
Comment (brambloemen) [inline nanomotif/binnary/utils.py:13]: Changed this function as I ran into another issue:

Traceback (most recent call last):
  File "/data/brbloemen/mambaforge/envs/nanomotif_411/bin/nanomotif", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/data/brbloemen/mambaforge/envs/nanomotif_411/lib/python3.11/site-packages/nanomotif/main.py", line 534, in main
    binnary(args)
  File "/data/brbloemen/mambaforge/envs/nanomotif_411/lib/python3.11/site-packages/nanomotif/main.py", line 426, in binnary
    contamination = detect_contamination.detect_contamination(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/brbloemen/mambaforge/envs/nanomotif_411/lib/python3.11/site-packages/nanomotif/binnary/detect_contamination.py", line 39, in detect_contamination
    contig_bin_comparison_score = split_bin_contig(contig_bin_comparison_score)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/brbloemen/mambaforge/envs/nanomotif_411/lib/python3.11/site-packages/nanomotif/binnary/utils.py", line 6, in split_bin_contig
    .with_columns(
     ^^^^^^^^^^^^^
  File "/data/brbloemen/mambaforge/envs/nanomotif_411/lib/python3.11/site-packages/polars/dataframe/frame.py", line 8270, in with_columns
    return self.lazy().with_columns(*exprs, **named_exprs).collect(_eager=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/brbloemen/mambaforge/envs/nanomotif_411/lib/python3.11/site-packages/polars/lazyframe/frame.py", line 1730, in collect
    return wrap_df(ldf.collect())
                   ^^^^^^^^^^^^^
polars.exceptions.ComputeError: dynamic pattern length in 'str.replace' expressions is not supported yet

Comment (SebastianDall) [reply nanomotif/binnary/utils.py:13]: Hi,
Thank you for contributing. Just out of curiosity which version of polars are you using?

Comment (brambloemen) [reply nanomotif/binnary/utils.py:13]: Hi, I used polars version 0.20.4 here, as I previously encountered the same issue in  #54

**Review by SorenHeidelbach — APPROVED** [review]

**Pull request merged**